packages/astro/src/webcontainer-files/filesmap.spec.ts

```diff
@@ -0,0 +1,271 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { afterEach, beforeEach, describe, expect, it, test, vi } from 'vitest';
+import { FilesMap } from './filesmap.js';
+
+const tmpDir = path.join(__dirname, `.tmp-${path.basename(__filename)}`);
+
+beforeEach(async () => {
+  await fs.rm(tmpDir, { force: true, recursive: true });
+  await fs.mkdir(tmpDir);
+});
+
+afterEach(async () => {
+  await fs.rm(tmpDir, { force: true, recursive: true });
+});
+
+describe('FilesMap', () => {
+  it('should initialize the graph of nodes by connecting them to one another', async () => {
+    await scaffoldTestFolders([
+      ['folder1', { file1: '' }],
+      ['folder2', { file2: '' }],
+    ]);
+
+    const folders = [path.join(tmpDir, 'folder1'), path.join(tmpDir, 'folder2')];
+    const logger = {
+      warn: vi.fn(),
+    };
+    const graph = await FilesMap.initGraph(folders, logger as any);
+
+    const node1 = graph.getFilesMapByFolder(folders[0]);
+    const node2 = graph.getFilesMapByFolder(folders[1]);
+
+    expect(node1).toBeDefined();
+    expect(node2).toBeDefined();
+    expect(logger.warn).not.toHaveBeenCalled();
+
+    expect(await node1?.toFiles(logger as any)).toMatchInlineSnapshot(`
+      {
+        "/file1": "",
+      }
+    `);
+    expect(await node2?.toFiles(logger as any)).toMatchInlineSnapshot(`
+      {
+        "/file2": "",
+      }
+    `);
+  });
```

Comment:
Tip: `expect()` doesn't do type narrowing but `assert()` does: https://vitest.dev/api/assert.html#assert-1

```suggestion
    assert(node1 !== undefined)
    assert(node2 !== undefined)
    expect(logger.warn).not.toHaveBeenCalled();

    expect(await node1.toFiles(logger as any)).toMatchInlineSnapshot(`
      {
        "/file1": "",
      }
    `);
    expect(await node2.toFiles(logger as any)).toMatchInlineSnapshot(`
      {
        "/file2": "",
      }
    `);
  });
```

Reply:
Oh that's super nice!! I'll update all the other tests